lib/std/fs.zig

```diff
@@ -771,6 +771,22 @@ pub const Dir = struct {
             });
         }
 
+        if (has_flock_open_flags and flags.lock_nonblocking) {
+            var fl_flags = os.fcntl(fd, os.F_GETFL, 0) catch |err| switch (err) {
```

Comment:
There should be no need to get fcntl flags for a freshly opened file descriptor.

Reply:
I wish it was this easy, on a Linux system you can see there's at least something like `O_WRONLY|O_LARGEFILE`.